model/src/application/component/mod.rs

```diff
@@ -0,0 +1,171 @@
+                    custom_id: "test custom id 2".into(),
+                    placeholder: Some("test placeholder".into()),
+                    min_values: Some(5),
+                    max_values: Some(25),
+                    options: Vec::from([SelectMenuOption {
+                        label: "test option label".into(),
+                        value: "test option value".into(),
+                        description: Some("test description".into()),
+                        emoji: None,
+                        default: false,
+                    }]),
+                }),
+            ]),
+        });
+
+        serde_test::assert_tokens(
+            &component,
+            &[
+                Token::Struct {
+                    name: "ActionRow",
+                    len: 2,
+                },
+                Token::Str("components"),
+                Token::Seq { len: Some(2) },
+                Token::Struct {
+                    name: "Button",
+                    len: 5,
+                },
+                Token::Str("custom_id"),
+                Token::Some,
+                Token::Str("test custom id"),
+                Token::Str("disabled"),
+                Token::Bool(false),
+                Token::Str("label"),
+                Token::Some,
+                Token::Str("test label"),
+                Token::Str("style"),
+                Token::U8(ButtonStyle::Primary as u8),
+                Token::Str("type"),
+                Token::U8(ComponentType::Button as u8),
+                Token::StructEnd,
+                Token::Struct {
+                    name: "SelectMenu",
+                    len: 6,
+                },
+                Token::Str("custom_id"),
+                Token::Str("test custom id 2"),
+                Token::Str("max_values"),
+                Token::Some,
+                Token::U8(25),
+                Token::Str("min_values"),
+                Token::Some,
+                Token::U8(5),
+                Token::Str("options"),
+                Token::Seq { len: Some(1) },
+                Token::Struct {
+                    name: "SelectMenuOption",
+                    len: 4,
+                },
+                Token::Str("default"),
+                Token::Bool(false),
+                Token::Str("description"),
+                Token::Some,
+                Token::Str("test description"),
+                Token::Str("label"),
+                Token::Str("test option label"),
+                Token::Str("value"),
+                Token::Str("test option value"),
+                Token::StructEnd,
+                Token::SeqEnd,
+                Token::Str("placeholder"),
+                Token::Some,
+                Token::Str("test placeholder"),
+                Token::Str("type"),
+                Token::U8(ComponentType::SelectMenu as u8),
+                Token::StructEnd,
+                Token::SeqEnd,
+                Token::Str("type"),
+                Token::U8(1),
```

Comment:
Shouldn't this be `Component::ActionRow as u8`?

Reply:
Done.